### PR TITLE
feat: add methods shareFile and unshareFile

### PR DESCRIPTION
**Pull request by gunar**
I would love for it to be `slack.files.share()` instead of `slack.shareFile()`
but I don't know how to do it.
Got bitten too many times by meta-programming.
Please change it if you find a reasonable way to do it.

Thanks!

### CODE REVIEW

**Comment by gunar** [comment]
blocker: https://github.com/invisible-tech/chance-extras/pull/3

File: src/slack/factory.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ const factory = ({ accessToken, teamId, cachedPaths = false, methods = false } =
   const existing = getSlack({ accessToken, teamId })
   if (existing) return existing
 
-  const unwrapped = (accessToken && teamId) ?
+  const unwrapped = accessToken ?
```
Comment (keenahn) [inline src/slack/factory.js]: Why? Some stuff depends on teamId, so we need it too

**Comment by keenahn** [comment]
Nice. It works. Should update the readme or at least put a comment that using these endpoints requires the browser token. 

Check my last commit. I removed the need for passing in the unwrapped slack, which I think is better.

File: src/slack/factory.js (hunk 1 above)
Comment (keenahn) [reply src/slack/factory.js]: nm, I agree with this change. I do need to get the stored api key by teamId for Osmosis, so I need it, but it's optional.

**Comment by gunar** [comment]
I wasn't able to make it work without unwrappedSlack -- seems fine now! working!

**Pull request merged**